dbms/src/Flash/tests/gtest_planner_interpreter.cpp

```diff
@@ -857,12 +857,14 @@ CreatingSets
      MockTableScan
  Union: <for test>
   Expression x 10: <final projection>
-   SharedQuery: <restore concurrency>
-    ParallelAggregating, max_threads: 10, final: true
-     Expression x 10: <remove useless column after join>
-      HashJoinProbe: <join probe, join_executor_id = Join_2>
-       Expression: <final projection>
-        MockTableScan)";
+   Expression: <cast after aggregation>
+    SharedQuery: <restore concurrency>
+     ParallelAggregating, max_threads: 10, final: true
+      Expression x 10: <before aggregation>
```

Comment:
projection action to remove useless column.
The output columns of join are {r_a, r_b, join_c, l_a, l_b, join_c}.
The columns agg needed are {r_a, join_c}

Reply:
Because now `PhysicalPlan::outputAndOptimize` can work on `Join`?

Reply:
yes, after optimize, Useless columns can be destructed earlier.